airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -218,31 +218,29 @@ private CancellationScope generateSyncWorkflowRunnable(final ConnectionUpdaterIn
         prepareForNextRunAndContinueAsNew(connectionUpdaterInput);
       }
 
-      // TODO (pedroslopez): The feature flags will actually be used in a future PR
-      Map<String, Boolean> featureFlags = getFeatureFlags(connectionUpdaterInput.getConnectionId());
+      final Map<String, Boolean> featureFlags = getFeatureFlags(connectionUpdaterInput.getConnectionId());
+      final boolean featureFlagCheckInputGeneration = featureFlags.getOrDefault(CheckInputGeneration.INSTANCE.getKey(), false);
```

Comment:
Feels like this should happen within an activity. My concern of about doing any top level feature flag evaluation is that with temporal replay, it could lead to inconsistencies. Looking up the flag once early and passing the value along should be more predictable.

Reply:
> Looking up the flag once early and passing the value along should be more predictable.

This is what we are doing! A new activity was introduced in https://github.com/airbytehq/airbyte-platform/pull/79 to retrieve feature flags for use in the workflow. This happens once at the start.

Reply:
getFeatureFlags actually runs the FeatureFlagFetchActivity

https://github.com/airbytehq/airbyte-platform/blob/7f567bb121f44154bb59584d9f9c9381171d265d/airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java#L652-L663

Reply:
Awesome!